src/coreComponents/physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.cpp

```diff
@@ -109,123 +109,354 @@ void EmbeddedSurfaceGenerator::initializePostSubGroups()
+  bool added = false;
+
+  if( ghostRank[tipElementIndex] < 0 )
+  {
+    //this tests if the element tipElementIndex is cut by the plane connecting currentTip to targetTip
+    isPositive = 0;
+    isNegative = 0;
+    for( localIndex kn = 0; kn < subRegion.numNodesPerElement(); kn++ )
+    {
+      nodeIndex = cellToNodes[tipElementIndex][kn];
+      LvArray::tensorOps::copy< 3 >( distVec, nodesCoord[nodeIndex] );
+      LvArray::tensorOps::subtract< 3 >( distVec, planeCenter );
+      // check if the dot product is zero
+      if( LvArray::tensorOps::AiBi< 3 >( distVec, normalVector ) > 0 )
+      {
+        isPositive = 1;
+      }
+      else if( LvArray::tensorOps::AiBi< 3 >( distVec, normalVector ) < 0 )
+      {
+        isNegative = 1;
+      }
+    } // end loop over nodes
+    if( isPositive * isNegative == 1 ) //if there are nodes on both sides of the fracture plane
+    {
+      //std::cout<<"plane name: "<<newFracturePlane->getName()<<std::endl;
+      added = embeddedSurfaceSubRegion.addNewEmbeddedSurface( tipElementIndex,
+                                                              er,
+                                                              esr,
+                                                              nodeManager,
+                                                              embSurfNodeManager,
+                                                              edgeManager,
+                                                              cellToEdges,
+                                                              newFracturePlane.get() );
+
+      if( added )
+      {
+        GEOSX_LOG_LEVEL_RANK_0( 2, "Element " << tipElementIndex << " is fractured" );
+        // Add the information to the CellElementSubRegion
+        subRegion.addFracturedElement( tipElementIndex, localNumberOfSurfaceElems );
+        newObjects.newElements[ {embeddedSurfaceRegion.getIndexInParent(), embeddedSurfaceSubRegion.getIndexInParent()} ].insert( localNumberOfSurfaceElems );
+        localNumberOfSurfaceElems++;
+      }
+    }
+  }
+
+
+  finiteElement::FiniteElementBase & subRegionFE = subRegion.template getReference< finiteElement::FiniteElementBase >( getDiscretizationName() );
+
+  finiteElement::dispatchlowOrder3D( subRegionFE, [&] ( auto const finiteElement )
+  {
+    using FE_TYPE = decltype( finiteElement );
+
+    auto kernel = CIcomputationKernel< FE_TYPE >( finiteElement,
+                                                  nodeManager,
+                                                  subRegion,
+                                                  embeddedSurfaceSubRegion );
+
+    using KERNEL_TYPE = decltype( kernel );
+
+    KERNEL_TYPE::template launchCIComputationKernel< parallelDevicePolicy< 32 >, KERNEL_TYPE >( kernel );
+  } );
+
 
   // add all new nodes to newObject list
-  for( localIndex ni = 0; ni < embSurfNodeManager.size(); ni++ )
+  // also, get index of edges that were just cut
+  array1d< globalIndex > newEdges;
```

Comment:
you can probably use an std::vector for this. And it would be nice to initialize the size if it's known.